e2e-test-server/endtoendserver/scenarios.go

```diff
@@ -77,33 +78,31 @@ func (s *Server) complexTraceHandler(ctx context.Context, req request, tracerPro
 	}
 
 	tracer := tracerProvider.Tracer(instrumentingModuleName)
+	ctx, span := tracer.Start(ctx, "complexTrace/root",
+		trace.WithAttributes(attribute.String(testIDKey, req.testID)))
+
 	func(ctx context.Context) {
-		ctx, span := tracer.Start(ctx, "complexTrace/root",
+		ctx, span := tracer.Start(ctx, "complexTrace/child1",
+			trace.WithSpanKind(trace.SpanKindServer),
 			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
-		defer span.End()
+		span.End()
 
 		func(ctx context.Context) {
-			ctx, span := tracer.Start(ctx, "complexTrace/child1",
-				trace.WithSpanKind(trace.SpanKindServer),
+			_, span := tracer.Start(ctx, "complexTrace/child2",
+				trace.WithSpanKind(trace.SpanKindClient),
 				trace.WithAttributes(attribute.String(testIDKey, req.testID)))
 			span.End()
-
-			func(ctx context.Context) {
-				_, span := tracer.Start(ctx, "complexTrace/child2",
-					trace.WithSpanKind(trace.SpanKindClient),
-					trace.WithAttributes(attribute.String(testIDKey, req.testID)))
-				span.End()
-			}(ctx)
 		}(ctx)
+	}(ctx)
 
-		func(ctx context.Context) {
-			_, span := tracer.Start(ctx, "complexTrace/child3",
-				trace.WithAttributes(attribute.String(testIDKey, req.testID)))
-			span.End()
-		}(ctx)
+	func(ctx context.Context) {
+		_, span := tracer.Start(ctx, "complexTrace/child3",
+			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
+		span.End()
 	}(ctx)
+	span.End()
 
-	return &response{statusCode: code.Code_OK}
+	return &response{statusCode: code.Code_OK, traceID: span.SpanContext().TraceID()}
```

Comment:
maybe this can fix the govet issue on golint.
```suggestion
	ctx, rootSpan := tracer.Start(ctx, "complexTrace/root",
		trace.WithAttributes(attribute.String(testIDKey, req.testID)))
	rootSpan.End()
	
	func(ctx context.Context) {
		ctx, span := tracer.Start(ctx, "complexTrace/child1",
			trace.WithSpanKind(trace.SpanKindServer),
			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
		span.End()

		func(ctx context.Context) {
			_, span := tracer.Start(ctx, "complexTrace/child2",
				trace.WithSpanKind(trace.SpanKindClient),
				trace.WithAttributes(attribute.String(testIDKey, req.testID)))
			span.End()
		}(ctx)
	}(ctx)

	func(ctx context.Context) {
		_, span := tracer.Start(ctx, "complexTrace/child3",
			trace.WithAttributes(attribute.String(testIDKey, req.testID)))
		span.End()
	}(ctx)

	return &response{statusCode: code.Code_OK, traceID: span.SpanContext().TraceID()}
```